### DISCUSSION TITLE
Introduce software metrics

### DISCUSSION THREAD

**Discussion by ThirtySomething** [microsoft/vscode-cpptools#7959] [Ideas]
Hello,

I would like to propose software metrics. [Jim Wanner](https://campwoodsw.com/emcomp/about-us/), the author of one of my favorite tools has ended his career as a software developer. With [SourceMonitor](https://www.campwoodsw.com/sourcemonitor.html), I was able to view various software metrics about source code. This is a great support to write better code. It is also a great help for restructuring and improving code. Unfortunately, there will be no further development for this tool.

So I had the idea that this could be a topic for the C++ extension.

A few metrics are easily implemented: The number of lines in the file, the number of lines of code, the number of comment lines, ... It gets more elaborate with the McCabe metric (cyclomatic complexity) or the branch depth. It is even more complex if this is to be done not only for a single file, but for a workspace. Then there are the average values and a maximum in addition to the previous values.

My first thought was to [propose ](https://github.com/microsoft/language-server-protocol/issues/1331) this as a feature of the language server protocol. However, due to my lack of knowledge about how VSCode and its components work, this was not the right place. The result is this post and I'm looking forward to the discussion.

Regards

Jochen

**Comment by Colengms** [comment]
Hi @ThirtySomething .  That sounds like a great idea.  Would you like to open an issue in this repo, as a feature request?  (We use GitHub issue to track bugs and feature requests).

**Reply by ThirtySomething** [reply]
Hello @Colengms ,

thanks for the hint, I've opened [this](https://github.com/microsoft/vscode-cpptools/issues/7974) issue.
